x-pack/plugins/infra/public/alerting/metric_threshold/components/custom_equation/metric_row_with_agg.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiSelect,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import React, { useMemo, useCallback } from 'react';
+import { get } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { Aggregators, CustomMetricAggTypes } from '../../../../../common/alerting/metrics';
+import { MetricRowControls } from './metric_row_controls';
+import { NormalizedFields, MetricRowBaseProps } from './types';
+
+interface MetricRowWithAggProps extends MetricRowBaseProps {
+  aggType?: CustomMetricAggTypes;
+  field?: string;
+  fields: NormalizedFields;
+}
+
+export const MetricRowWithAgg: React.FC<MetricRowWithAggProps> = ({
```

Comment:
nit: Since we're not using `children`, I guess we could remove the usage of `React.FC` and just go for a functional component. WDYT?

```ts
const MetricRowWithAgg = ({
 name,
  aggType = Aggregators.AVERAGE,
  field,
  onDelete,
  disableDelete,
  fields,
  aggregationTypes,
  onChange,
  errors,
}: MetricRowWithAggProps) => 
```